File: desktop/src/native/messages/contacts.ts
/* eslint-disable camelcase */

export interface Contact {
  id: string,
  created_at: Date,
  updated_at: Date,
  birthday: Date | null,
  first_name: string,
  last_name: string,
  notes: string,
  emails: Email[],
  phones: Phone[],
  websites: Website[],
  organizations: Organization[],
  addresses: Address[],
  device_id: string,
}

export interface Email {
  email: string,
  label: string,
}

export interface Phone {
  phone: string,
  label: string,
}

export interface Website {
  website: string,
  label: string,
}

export interface Organization {
  name: string,
  title: string,
}

export interface Address {
  city: string,
  country: string,
  label: string,
  postal_code: string,
  street_address: string,
  street_address2: string,
}

export interface GuiContact {
  contact: Contact,
}

export interface GuiContacts {
  contacts: Contact[],
}

export interface GuiCreateContact {
  birthday: Date | null,
  first_name: string,
  last_name: string,
  notes: string,
  emails: Email[],
  phones: Phone[],
  websites: Website[],
  organizations: Organization[],
  addresses: Address[],
  device_id: string,
}

export interface GuiDeleteContact {
  id: string,
}

File: core/bloom/messages/src/contacts.rs
use super::{from_message, Message};
use bloom_models::contacts;
use serde::{Deserialize, Serialize};

////////////////////////////////////////////////////////////////////////////////////////////////////
/// GUI
////////////////////////////////////////////////////////////////////////////////////////////////////

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct GuiListContacts {}
from_message!(GuiListContacts, Message::ContactsGuiListContacts);

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct GuiContact {
    pub contact: contacts::db::Conatct,
}
from_message!(Box<GuiContact>, Message::ContactsGuiContact);

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct GuiContacts {
    pub contacts: Vec<contacts::db::Conatct>,
}
from_message!(GuiContacts, Message::ContactsGuiContacts);

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct GuiCreateContact {
    pub first_name: String,
    pub last_name: String,
    pub notes: String,
    pub addresses: serde_json::Value,
    pub birthday: Option<chrono::DateTime<chrono::Utc>>,
    pub organizations: serde_json::Value,
    pub emails: serde_json::Value,
    pub phones: serde_json::Value,
    pub websites: serde_json::Value,
    pub device_id: String,
}
from_message!(Box<GuiCreateContact>, Message::ContactsGuiCreateContact);

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct GuiDeleteContact {
    pub id: String,
}
from_message!(GuiDeleteContact, Message::ContactsGuiDeleteContact);

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct GuiUpdateContact {
    pub contact: contacts::db::Conatct,
}
from_message!(Box<GuiUpdateContact>, Message::ContactsGuiUpdateContact);
